components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/bean/RewriteContext.java

```diff
@@ -17,28 +17,50 @@
  */
 package org.wso2.carbon.identity.context.rewrite.bean;
 
+import java.util.regex.Pattern;
+
 public class RewriteContext {
+
     private boolean isWebApp;
+
     private String context;
 
+    private Pattern pattern;
+
     public RewriteContext(boolean isWebApp, String context) {
+
         this.isWebApp = isWebApp;
         this.context = context;
+        this.pattern = Pattern.compile("/t/([^/]+)" + context);
```

Comment:
This will be initialized per request, right? Pattern compile is an operation that creates a high GC pressure. Is there any possibility to avoid initializing this per request?

Reply:
This is called only during the Valve startup https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/100/files#diff-2b30376572fa3cfee6f27079baf5a0e7R61